bot/src/main/kotlin/bot/horo/bot/command/ProfileExtension.kt

```diff
@@ -0,0 +1,145 @@
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            publicSubCommand(::ProfileArguments) {
+                name = "view.profile.sub.name"
+                description = "view.profile.sub.description"
+
+                action {
+                    //access the ID from the parsed arguments
+                    val id = arguments.target.id
+
+                    //use the user ID to get the user from the guild
+                    val guild = this@action.guild!!
+                    val user = guild.getMember(id)
+
+                    val rolesLabel = translate("view.profile.sub.field.roles")
+
+                    respond {
+                        embed {
+                            if (user.avatar == null) {
+                                thumbnail {
+                                    url = user.defaultAvatar.url
+                                }
+                                author {
+                                    icon = user.defaultAvatar.url
+                                    name = user.tag
+                                }
+                            } else {
+                                thumbnail {
+                                    url = user.avatar!!.url
+                                }
+                                author {
+                                    icon = user.avatar!!.url
+                                    name = user.tag
+                                }
+                            }
+                            field {
+                                name = translate("view.profile.sub.field.nickname")
+                                value = user.mention
+                            }
+                            field {
+                                name = translate("view.profile.sub.field.joinedAt")
+                                value = user.joinedAt.toMessageFormat()
+                            }
+                            if (user.roleIds.isNotEmpty()) {
+                                field {
+                                    name = "$rolesLabel (${user.roleIds.size})"
+                                    value = user.roleIds.joinToString(" ") { "<@&$it>" }
+                                }
+                            } else {
+                                field {
+                                    name = "$rolesLabel (0)"
+                                    value = "This user has no roles"
+                                }
+                            }
+                            field {
+                                name = translate("view.profile.sub.field.createdAt")
+                                value = user.createdAt.toMessageFormat()
+                            }
+                            footer {
+                                text = "ID:" + user.id.toString()
```

Comment:
Not translated, probably missing a space, and should probably also be a template string to avoid the `toString` call.

Reply:
Pretty sure you don't need to  call toString on Snowflake types anyway. Needs to be moved into string literal though.